server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sRequestToRawBody.scala

```diff
@@ -29,7 +28,7 @@ private[http4s] class Http4sRequestToRawBody[F[_]: Sync: ContextShift](serverOpt
         // TODO: use MultipartDecoder.mixedMultipart once available?
         implicitly[EntityDecoder[F, multipart.Multipart[F]]].decode(req, strict = false).value.flatMap {
           case Left(failure) =>
-            throw new IllegalArgumentException("Cannot decode multipart body: " + failure) // TODO
+            Sync[F].pure(failure.asInstanceOf[R])
```

Comment:
So here we're cheating, as this isn't really an instance of `R`, right (even though Scala can't prove it, we can ;) ). Maybe we could return a failed effect here, and add a `.recover` call which would handle `MessageBodyFailure` in `EndpointToHttp4sServer`?